### PR TITLE
update sequence diagrams based on feedback

### PR DESCRIPTION
**Pull request by maryamtahhan**
Edit the sequence diagrams based on feedback from the review session. I can't find others to add them as reviewers. Will post to slack to bring to their attention.

### CODE REVIEW

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 1]
```diff
@@ -17,28 +15,33 @@ The main idea behind this document is to specify the flow of VF/SF allocation fr
 - Service metrics will be collectable through services that are fully integrated
 as a Kubernetes deployment.
 
-- A third cluster will be used as the broker.
-
 - The Kublet will be host networked.
 
+- The infrastructure is the `primary` entity driving/making networking decisions on behalf of the host.
```
Comment (maryamtahhan) [inline sequence-diagrams/host-provisioning-xPU-VF.md:20]:
```suggestion
- The infrastructure (AKA the xPU) is the `primary` entity driving/making networking decisions on behalf of the host.
```

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 2]
```diff
@@ -17,28 +15,33 @@ The main idea behind this document is to specify the flow of VF/SF allocation fr
...
-- If the infra cluster needs access to many Kubernetes resources in the tenant cluster, does it make sense to have a broker or is direct cluster access a better approach.
+- If the infra cluster needs access to many Kubernetes resources in the tenant cluster, does it make sense to have a broker or is direct cluster access a better approach and does a broker architecture scale well across clusters when many of the resources that need to be shared are Kubernetes resources.
 - Should we even consider using something like microshift on the DPU in the two cluster scenario.
 - Need to think about how nodeport will work.
+- How to use CNI in hetrogenous deployments.
+
+### To Broker or not to Broker?
+
+A **Broker** is an entity that syncs OPI CRDs between the Tenant and Infra clusters. It's a singleton component that is deployed on a cluster whose Kubernetes API must be accessible by all of the participating clusters. The Broker cluster may be one of the participating clusters or a standalone cluster without the other components deployed. The Agent components deployed in each participating cluster are configured with the information to securely connect to the Broker cluster’s API.
```
Comment (maryamtahhan) [inline sequence-diagrams/host-provisioning-xPU-VF.md:31]:
```suggestion
A **Broker** is an entity that could be used to sync OPI CRDs between the Tenant and Infra clusters. It's a singleton component that is deployed on a cluster whose Kubernetes API must be accessible by all of the participating clusters. The Broker cluster may be one of the participating clusters or a standalone cluster without the other components deployed. The Agent components deployed in each participating cluster are configured with the information to securely connect to the Broker cluster’s API.
```

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 3]
```diff
@@ -17,28 +15,33 @@ The main idea behind this document is to specify the flow of VF/SF allocation fr
...
 
-### Provisioning via CRDs
+The xPU Kubernetes architecture will avoid leveraging a Broker within the xPU as the infra cluster and the tenant cluster are tightly coupled. The infra cluster will need a access to several Kubernetes resources and it does not make sense to replicate these (resources) through CRDs via a Broker. There's also a question of latency introduced by the Broker and whether or not it can scale.
```
Comment (maryamtahhan) [inline sequence-diagrams/host-provisioning-xPU-VF.md:33]: Mark this as a proposal

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 4]
```diff
@@ -47,51 +50,44 @@ as a Kubernetes deployment.
...
-  - Syncs OPI CRDs between the Tenant and Infra clusters.
-  - The Broker is a singleton component that is deployed on a cluster whose Kubernetes API must be accessible by all of the participating clusters.
-  - The Broker cluster may be one of the participating clusters or a standalone cluster without the other components deployed. The Agent components deployed in each participating cluster are configured with the information to securely connect to the Broker cluster’s API.
 
-The following high level diagram provides an overview of the entities working
-together to provision a primary network interface for a pod.
+### Required Resources
 
-![primary-int-provisioning-highlevel](./images/primary-int-provisioning-highlevel.png)
+TODO (WIP)
```
Comment (maryamtahhan) [inline sequence-diagrams/host-provisioning-xPU-VF.md:56]: This is a non exhaustive list of resources that need to be shared between a tenant and infra cluster

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 5]
```diff
@@ -4,11 +4,9 @@ The main idea behind this document is to specify the flow of VF/SF allocation fr
```
Comment (maryamtahhan) [inline sequence-diagrams/host-provisioning-xPU-VF.md:4]: mention that the focus of this document will be the 2 cluster scenario

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 6]
```diff
@@ -47,51 +50,44 @@ as a Kubernetes deployment.
...
-- EndpointSlice: reflecting the endpoint slices in the tenant cluster.
-- Network policy
-
-### Single Cluster
+The following high level diagram provides an overview of the entities working
+together to provision a secondary network interface for a pod.
 
-The following sequence diagram provides an overview of the entities working
-together to provision an interface for a pod in the single cluster scenario.
+TODO
```
Comment (maryamtahhan) [inline sequence-diagrams/host-provisioning-xPU-VF.md:78]: I'm not sure this looks different to the previous diagram

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 7]
```diff
@@ -47,54 +48,73 @@ as a Kubernetes deployment.
...
+    Subnets []string `json:"subnets"` // or maybe other routing info?
+    IP string `json:"ip"`
+    Backend string `json:"backend,omitempty"` //subfunction or virtual function
+    Status NetdevStatus `json:"status"`
+}
+
+type Netdev struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec NetdevSpec `json:"spec"`
```
Comment (Billy99) [inline sequence-diagrams/host-provisioning-xPU-VF.md:116]: Probably want status as a separate structure, that way Host owns Spec and DPU owns Status. Question is whether IP belongs in Spec or Status (depending on whether Host or DPU writes it).
```suggestion
    Spec NetdevSpec `json:"spec"`
    // +optional
    Status NetdevStatus `json:"status,omitempty"`
```

File: sequence-diagrams/host-provisioning-xPU-VF.md [hunk 8]
```diff
@@ -47,54 +48,73 @@ as a Kubernetes deployment.
...
+
+const (
+    Connected NetdevStatus = "connected"
+    Connecting NetdevStatus = "connecting"
+    Disconnected NetdevStatus = "disconnected"
+    ConnectionError NetdevStatus = "error"
+)
+
+type NetdevSpec struct {
+    NetdevID string `json:"netdev_id"` // SerialNum_PFID_SFID
```
Comment (Billy99) [inline sequence-diagrams/host-provisioning-xPU-VF.md:105]: Purely for reference, this is what is passed in annotations in OVN-Kubernetes (written as a map indexed by the namespace):
```
type DPUConnectionDetails struct {
	PfId         string `json:"pfId"`
	VfId         string `json:"vfId"`
	SandboxId    string `json:"sandboxId"`
	VfNetdevName string `json:"vfNetdevName,omitempty"`
}
```


The IP info was already in the Pod Annotation, so it is not part of the DPU specific content.
```
// Internal struct used to marshal PodAnnotation to the pod annotation
type podAnnotation struct {
	IPs      []string   `json:"ip_addresses"`
	MAC      string     `json:"mac_address"`
	Gateways []string   `json:"gateway_ips,omitempty"`
	Routes   []podRoute `json:"routes,omitempty"`

	IP      string `json:"ip_address,omitempty"`
	Gateway string `json:"gateway_ip,omitempty"`
}

// Internal struct used to marshal PodRoute to the pod annotation
type podRoute struct {
	Dest    string `json:"dest"`
	NextHop string `json:"nextHop"`
}
```

File: sequence-diagrams/host-provisioning-xPU-VF.md (hunk 7 above)
Comment (maryamtahhan) [reply sequence-diagrams/host-provisioning-xPU-VF.md:116]: makes sense

File: sequence-diagrams/host-provisioning-xPU-VF.md (hunk 8 above)
Comment (Billy99) [reply sequence-diagrams/host-provisioning-xPU-VF.md:105]: I forgot to mention, for IP, if I recall, `ip_address` and `gateway_ip` were there originally, and then when dual stack was added, it became an array of address, `ip_addresses` and `gateway_ips`. The original was left around for backwards compatibility.

**Pull request closed without merging**